apps/web/src/entry/ssr.tsx

```
@@ -0,0 +1,43 @@
+
+import type { TreeRouter } from "./router";
+import { createRouter } from "./router";
+
+const getExternalContext = (): ExternalRouterContext => {
+	const cookies = parseCookies();
+	const initialValues = getStoreValuesFromInitialValues(cookies);
+	return {
+		initialValues,
+		// SSR always has web request
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		environment: { type: "server", request: getWebRequest()! },
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)